robot/impl/local_robot.go

```diff
@@ -677,34 +693,52 @@ func (r *localRobot) updateWeakDependents(ctx context.Context) {
 		}
 	}
 
+	timeout := r.getTimeout()
 	// NOTE(erd): this is intentionally hard coded since these services are treated specially with
 	// how they request dependencies or consume the robot's config. We should make an effort to
 	// formalize these as servcices that while internal, obey the reconfigure lifecycle.
 	// For example, the framesystem should depend on all input enabled components while the web
 	// service depends on all resources.
 	// For now, we pass all resources and empty configs.
 	for resName, res := range internalResources {
-		switch resName {
-		case web.InternalServiceName:
-			if err := res.Reconfigure(ctx, allResources, resource.Config{}); err != nil {
-				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
-			}
-		case framesystem.InternalServiceName:
-			fsCfg, err := r.FrameSystemConfig(ctx)
-			if err != nil {
-				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
-				continue
-			}
-			if err := res.Reconfigure(ctx, components, resource.Config{ConvertedAttributes: fsCfg}); err != nil {
-				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
+		resChan := make(chan struct{}, 1)
+		resName := resName
+		res := res
+		ctxWithTimeout, timeoutCancel := context.WithTimeout(ctx, timeout)
```

Comment:
we'll want to wrap this context + go call in an "immediately called anonymous function" so that we don't add a bunch of defers to the stack that only execute once the updateWeakDependents is done.

Reply:
Or you can refactor this into a smaller function here